controllers/dataprotection/restore_controller_test.go

```diff
@@ -245,6 +245,40 @@ var _ = Describe("Restore Controller test", func() {
 					}, nil)
 			})
 
+			It("test restore is failed when check failed in new action", func() {
+				By("expect for restore is Failed and BackupRepoFailed message")
+				restore := initResourcesAndWaitRestore(true, false, true, dpv1alpha1.RestorePhaseFailed,
+					func(f *testdp.MockRestoreFactory) {
+						f.Get().Spec.Backup.Name = "wrongBackup"
+					}, nil)
+				Eventually(testapps.CheckObj(&testCtx, client.ObjectKeyFromObject(restore), func(g Gomega, r *dpv1alpha1.Restore) {
+					exist := false
```

Comment:
Use [metav1.IsStatusConditionTrue()](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/meta#IsStatusConditionTrue) to simply the code.